### ISSUE TITLE
Add GraphQL endpoint

### ISSUE THREAD

**Issue by yozlet** [simonw/datasette#176]
Would make it much easier to build React & similar frontends. Maybe with https://github.com/graphql-python/sanic-graphql ?

**Comment by wulfmann** [comment]
This project probably would not be the place for that. This is a layer for sqllite specifically. It solves a similar problem as graphql, so adding that here wouldn't make sense.

Here's an example i found from google that uses micro to run a graphql microservice. you'd just then need to connect your db.
https://github.com/timneutkens/micro-graphql

**Comment by yozlet** [comment]
@wulfmann I think I disagree, except I'm not entirely sure what you mean by that first paragraph. The JSON API that Datasette currently exposes is quite different to GraphQL.

Furthermore, there's no "just" about connecting micro-graphql to a DB; at least, no more "just" than adding any other API. You still need to configure the schema, which is exactly the kind of thing that Datasette does for JSON API. This is why I think that GraphQL's a good fit here.

**Comment by wulfmann** [comment]
@yozlet Yes I think that I was confused when I posted my original comment. I see your main point now and am in agreement.

**Comment by gijs** [comment]
👍 I'd like this too!

**Comment by wuhland** [comment]
great idea!

**Comment by eads** [comment]
👋 I'd be interested in building this out in Q1 or Q2 of 2020 if nobody has tackled it by then. I would love to integrate Datasette into @thechicagoreporter's practice, but we're also fully committed to GraphQL moving forward.

**Comment by nkirsch** [comment]
@eads I'm interested in helping, if there's still a need...

**Comment by simonw** [comment]
This happened in the https://github.com/simonw/datasette-graphql plugin.
